test/module/ecdsa-registry-module.specs.ts

```diff
@@ -0,0 +1,156 @@
+            await SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            const userSA = await hre.ethers.getContractAt("SmartAccount", expectedSmartAccountAddress);
+
+            expect( await EcdsaRegistryModule.smartAccountOwners(userSA.address)).to.be.equal(await smartAccountOwner.getAddress());
+
+        });
+
+        // Pass in non(EOA) as owner in the setupData and check for revert
+        it("Reverts when trying to set Smart Contract as owner of the Smart Account", async()=>{
+            const {SAFactory,EcdsaRegistryModule} = await setupTests();
+
+            const randomContractCode = `
+            contract random {
+                function returnAddress() public view returns(address){
+                    return address(this);
+                }
+            }
+            `;
+            const randomContract = await deployContract(deployer,randomContractCode);
+
+
+            let EcdsaOwnershipSetupData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [await randomContract.address]
+              );
+
+            // Getting the address of the to be deployed SA does not get revert as it's just a bunch of abi.encode operations
+            const expectedSmartAccountAddress = await SAFactory.getAddressForCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            // This should revert
+            await expect(SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex)).to.reverted;
+            // Extra checks
+            // smartAccountOwners mapping should point to address(0)
+            expect( await EcdsaRegistryModule.smartAccountOwners(expectedSmartAccountAddress)).to.be.equal(AddressZero);
+        });
+
+        // Listen for AlreadyInitedForSmartAccount(msg.sender) error
+        it("Reverts when calling again after initialization", async()=>{
+            const {SAFactory,EcdsaRegistryModule,entryPoint} = await setupTests();
+
+            let EcdsaOwnershipSetupData = EcdsaRegistryModule.interface.encodeFunctionData(
+                "initForSmartAccount",
+                [await smartAccountOwner.getAddress()]
+              );
+
+            const expectedSmartAccountAddress = await SAFactory.getAddressForCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            await SAFactory.deployCounterFactualAccount(EcdsaRegistryModule.address,EcdsaOwnershipSetupData,smartAccountDeploymentIndex);
+            const userSA = await hre.ethers.getContractAt("SmartAccount", expectedSmartAccountAddress);
+
+            await deployer.sendTransaction({
```

Comment:
You better fund userSA in the setupTests once, rather than making it inside every test case